mlflow/store/artifact/http_artifact_repo.py

```diff
@@ -76,3 +81,41 @@ def delete_artifacts(self, artifact_path=None):
         endpoint = posixpath.join("/", artifact_path) if artifact_path else "/"
         resp = http_request(self._host_creds, endpoint, "DELETE", stream=True)
         augmented_raise_for_status(resp)
+
+    def create_multipart_upload(self, local_file, num_parts=1, artifact_path=None):
```

Comment:
One challenge I found in my design is the client doesn't know the underlying artifact repository in the server.

Reply:
Let's say we have a large file. To upload this file using MPU, the client first needs to ask the server "hey, can you do MPU?".

Reply:
I think we are going to implement mpu for azure, do you mean for other stores like ftp or hdfs?

Also, it seems unavoidable to have to "ask" the server, unless we decide to support mlflow server side mpu (i.e., option 1 in [Link](https://docs.google.com/document/d/19dqSa9YWhW7MQ0fn_wdjR8JecO1dYwxyrPE5GBaz8jA?usp=sharing)) for ftp, hdfs, etc.

Reply:
> do you mean for other stores like ftp or hdfs?

Yes, other stores that we won't support MPU for.

Reply:
1. Check the file size.
2. If it's larger than a threshold (say 500 MB), attempt to start MPU.
3. If the server responds with `501` (not implemented), then fall back.

Reply:
from my point of view, the user should be able to choose whether to use mpu or not, and the threshold seems a bit arbitrary (and possibly bounded by storage provider requirements?)

how about we expose another endpoint like `/perform-mpu?normal_if_failed=True` that does exactly what you said? so that the user only needs to call that one endpoint to do everything. but if the user put `normal_if_failed=False`, then the request will fail if mpu is not supported. 

this way, the user still has access to the more atomic `create`, `abort` & `complete` actions